pkg/controllers/binding/common.go

```diff
@@ -193,6 +197,31 @@ func RecordAppliedOverrides(cops *overridemanager.AppliedOverrides, ops *overrid
 	return annotations, nil
 }
 
+// mergeConflictResolution determine the conflictResolution annotation of Work: preferentially inherit from RT, then RB
+func mergeConflictResolution(workload *unstructured.Unstructured, conflictResolutionInBinding policyv1alpha1.ConflictResolution,
+	annotations map[string]string) map[string]string {
+	// conflictResolutionInRT refer to the annotation in ResourceTemplate
+	conflictResolutionInRT := util.GetAnnotationValue(workload.GetAnnotations(), workv1alpha2.ResourceConflictResolutionAnnotation)
+
+	// the final conflictResolution annotation value of Work inherit from RT preferentially
+	// so if conflictResolution annotation is defined in RT already, just copy the value and return
+	if conflictResolutionInRT == workv1alpha2.ResourceConflictResolutionOverwrite || conflictResolutionInRT == workv1alpha2.ResourceConflictResolutionAbort {
+		annotations[workv1alpha2.ResourceConflictResolutionAnnotation] = conflictResolutionInRT
+		return annotations
+	} else if conflictResolutionInRT != "" {
+		// ignore its value and add logs if conflictResolutionInRT is neither abort nor overwrite.
+		klog.Errorf("ignore the invalid resource conflict-resolution annotation in ResourceTemplate: %s", conflictResolutionInRT)
+	}
+
+	if conflictResolutionInBinding == policyv1alpha1.ConflictOverwrite {
+		annotations[workv1alpha2.ResourceConflictResolutionAnnotation] = workv1alpha2.ResourceConflictResolutionOverwrite
+	} else {
```

Comment:
return above and don't need else.

Reply:
This line was specially added by me. Supposing `ResourceTemplate`'s annotation is "", while `PP`'s conflictResolution is "overwrite", my expected result is "overwrite". If directly return above without `else`, the result annotation in `workload` will be "", which means "abort".

![image](https://github.com/karmada-io/karmada/assets/30589999/fbaec5f6-d8b7-4628-88dc-934977d80d21)

Reply:
I got what you mean, you just want to reduce the complexity of the loop, thks.